lib/routes/password.js

```diff
@@ -143,56 +145,155 @@ module.exports = function (
         var authPW = Buffer(request.payload.authPW, 'hex')
         var wrapKb = Buffer(request.payload.wrapKb, 'hex')
         var authSalt = crypto.randomBytes(32)
+        var sessionTokenId = request.payload.sessionToken
         var password = new Password(authPW, authSalt, verifierVersion)
-        db.deletePasswordChangeToken(passwordChangeToken)
-          .then(
-            function () {
-              return password.verifyHash()
-                .then(
-                  function (verifyHash) {
-                    return password.wrap(wrapKb)
-                      .then(
-                        function (wrapWrapKb) {
-                          return db.resetAccount(
-                            passwordChangeToken,
-                            {
-                              verifyHash: verifyHash,
-                              authSalt: authSalt,
-                              wrapWrapKb: wrapWrapKb,
-                              verifierVersion: password.version
-                            }
-                          )
-                        }
-                      )
+        var wantsKeys = (request.query.keys === 'true')
+        var account, verifyHash, sessionToken, keyFetchToken, verifiedStatus
+
+        if (!sessionTokenId) {
+          // If no sessionToken, this could be a legacy client
+          // attempting to change password, process request as legacy client
+          changePassword()
```

Comment:
Based on vidyo, we decided to add sessionToken to request body.

https://github.com/mozilla/fxa-auth-server/blob/b733e79813c1b6f2661f9735a3990c9b4f94adef/lib/routes/account.js#L1268